packages/eslint-plugin/tests/rules/require-await.test.ts

```diff
@@ -667,6 +939,169 @@ for await (let num of asyncIterable) {
         {
           messageId: 'missingAwait',
           data: { name: 'Async arrow function' },
+          suggestions: [
+            {
+              messageId: 'removeAsync',
+              output: `
+        async function foo() {
+          await (() => {
+            doSomething();
+          });
+        }
+      `,
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+        const obj = {
+          async: async function foo() {
+            bar();
+          },
+        };
+      `,
+      errors: [
+        {
+          messageId: 'missingAwait',
+          data: { name: "Async method 'async'" },
+          suggestions: [
+            {
+              output: `
+        const obj = {
+          async: function foo() {
+            bar();
+          },
+        };
+      `,
+              messageId: 'removeAsync',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      // This test verifies that the async keyword and any following
+      // whitespace is removed, but not the following comments.
+      // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+      code: `
+        async    /* test */ function foo() {
+          doSomething();
+        }
+      `,
+      errors: [
+        {
+          messageId: 'missingAwait',
+          data: { name: "Async function 'foo'" },
+          suggestions: [
+            {
+              output: `
+        /* test */ function foo() {
+          doSomething();
+        }
+      `,
+              messageId: 'removeAsync',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+        class A {
+          a = 0;
+          async [b]() {
+            return 0;
+          }
+        }
+      `,
+      languageOptions: {
+        parserOptions: { ecmaVersion: 2022 },
+      },
+      errors: [
+        {
+          messageId: 'missingAwait',
+          data: { name: 'Async method' },
+          suggestions: [
+            {
+              output: `
+        class A {
+          a = 0;
+          [b]() {
+            return 0;
+          }
+        }
+      `,
+              messageId: 'removeAsync',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      // This test must not have a semicolon after "foo".
+      // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+      code: `
```

Comment:
nit - we have a `noFormat` tag we like to use for intentionally unformatted test cases 
```suggestion
      code: noFormat`
```